### PR TITLE
Allow specifying version label at initial install

### PR DESCRIPTION
**Pull request by divolgin**
#### What type of PR is this?

type::feature
#### What this PR does / why we need it:

This allows specifying an app version label during the initial license install.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
* Added support for installing a specific application release.
```

#### Does this PR require documentation?

https://github.com/replicatedhq/replicated-docs/pull/232

### CODE REVIEW

File: pkg/upstream/replicated.go [hunk 1]
```diff
@@ -448,6 +462,7 @@ func readReplicatedAppFromLocalPath(localPath string, localCursor ReplicatedCurs
 }
 
 func downloadReplicatedApp(replicatedUpstream *ReplicatedUpstream, license *kotsv1beta1.License, cursor ReplicatedCursor, reportingInfo *reportingtypes.ReportingInfo) (*Release, error) {
+	debug.PrintStack()
```
Comment (sgalsaleh) [inline pkg/upstream/replicated.go:465]: Cleanup

**Comment by sgalsaleh** [comment]
How does this work for airgap?

**Comment by divolgin** [comment]
This feature cannot be supported for airgap.  The best we can do is check label in the bundle and reject it if it's wrong.

File: cmd/kots/cli/install.go [hunk 2]
```diff
@@ -429,6 +430,7 @@ func InstallCmd() *cobra.Command {
 	cmd.Flags().Bool("disable-image-push", false, "set to true to disable images from being pushed to private registry")
 	cmd.Flags().Bool("strict-security-context", false, "set to explicitly enable explicit security contexts for all kots pods and containers (may not work for some storage providers)")
 	cmd.Flags().Bool("skip-compatibility-check", false, "set to true to skip compatibility checks between the current kots version and the app")
+	cmd.Flags().String("app-version-label", "", "application version label. if not specified, the latest version will be installed")
```
Comment (sgalsaleh) [inline cmd/kots/cli/install.go:433]:
```suggestion
	cmd.Flags().String("app-version-label", "", "application version label to be installed. if not specified, the latest version will be installed")
```

**Review by sgalsaleh — APPROVED** [review]

**Review by sgalsaleh — DISMISSED** [review]

File: pkg/airgap/airgap.go [hunk 3]
```diff
@@ -317,6 +317,11 @@ func CreateAppFromAirgap(opts CreateAirgapAppOpts) (finalError error) {
 		}
 	}
 
+	err = kotsutil.RemoveAppVersionLabelFromInstallationParams(kotsadmtypes.KotsadmConfigMap)
```
Comment (sgalsaleh) [inline pkg/airgap/airgap.go:320]: should be done for online too

Comment (sgalsaleh) [inline pkg/airgap/airgap.go:320]: should do the same for online

Comment (divolgin) [reply pkg/airgap/airgap.go:320]: good call. updated.

File: pkg/handlers/license.go [hunk 4]
```diff
@@ -338,26 +338,35 @@ func (h *Handler) UploadNewLicense(w http.ResponseWriter, r *http.Request) {
...
 		kotsKinds, err := online.CreateAppFromOnline(createAppOpts)
 		if err != nil {
 			logger.Error(err)
 			uploadLicenseResponse.Error = err.Error()
-			JSON(w, 500, uploadLicenseResponse)
+			JSON(w, http.StatusInternalServerError, uploadLicenseResponse)
+			return
+		}
+
+		err = kotsutil.RemoveAppVersionLabelFromInstallationParams(kotsadmtypes.KotsadmConfigMap)
```
Comment (sgalsaleh) [inline pkg/handlers/license.go:365]: why not in `CreateAppFromOnline`? since that's called from multiple places, like resuming an online install.

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**